controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/DeploymentMetricsMaintainer.java

```diff
@@ -36,22 +37,24 @@ protected void maintain() {
                     DeploymentMetrics appMetrics = new DeploymentMetrics(metrics.queriesPerSecond(), metrics.writesPerSecond(),
                                                                          metrics.documentCount(), metrics.queryLatencyMillis(), metrics.writeLatencyMillis());
 
-                    // Avoid locking for a long time, due to slow YAMAS.
                     try (Lock lock = controller().applications().lock(application.id())) {
-                        // Deployment (or application) may have changed (or be gone) now:
-                        controller().applications().get(application.id()).ifPresent(freshApplication -> {
-                            Deployment freshDeployment = freshApplication.deployments().get(deployment.zone());
-                            if (freshDeployment != null)
-                                controller().applications().store(freshApplication.with(freshDeployment.withMetrics(appMetrics)), lock);
-                        });
+
+                        // Deployment or application may have changed (or be gone) now:
+                        application = controller().applications().get(application.id()).orElse(null);
+                        if (application == null)
```

Comment:
`get` returns an optional. Why not use `isPresent` instead of converting it to a `null`?

Reply:
Agree.

Reply:
Because I would have to introduce an additional local variable which would should represent the same as `application`, but with a different name. I think this confuses more than it clarifies; it may increase the risk of using the wrong variable further down; it makes the code less uniform; and it makes for quite a bit more text. I considered it, but didn't like it.

Reply:
Feel free to explain my error or bad taste.

Reply:
Not sure I understand. Can't you just use `get()` on the optional in the places you use the application?

Reply:
Martin: Because there is an "application" variable already.
Jonm: Good point. LGTM